### ISSUE TITLE
gitleaksignore fingerprint doesn't play well with github's super linter

### ISSUE THREAD

**Issue by jasonkarns** [zricethezav/gitleaks#1059]
**Is your feature request related to a problem? Please describe.**

gitleaks is a linter that is bundled with (and enabled by default in) GitHub's SuperLinter action. https://github.com/github/super-linter#supported-linters

When gitleaks (via superlinter) finds a secreat, the fingerpint uses an absolute path for its filename:

`/github/workspace/REPO_ROOT/PATH/TO/FILE:private-key:1`

However, that filepath is only relevant to runs of gitleaks _from github actions_.

Adding a relative-path pattern to `.gitleaksignore` does not properly allow the secret:

`PATH/TO/FILE:private-key:1`

**Describe the solution you'd like**

I'd like there to be a way to easily ignore intentionally-committed secrets using .gitleaksignore in a manner that allows gitleaks to be run both from developer's local machine and from github actions.

**Describe alternatives you've considered**

Configuration via the .toml is one option. However, one of the primary desires of using superlinter is to not need a dozen configuration files.

**Additional context**

cc @zricethezav

**Comment by Sumtin** [comment]
Running into very similar issue.  I'm running Gitleaks as part of an Azure DevOps Pipeline Decorator - a [decorator ](https://learn.microsoft.com/en-us/azure/devops/extend/develop/add-pipeline-decorator?view=azure-devops ) is just a private extension.  I'm also using self-hosted build agents.

Under this setup I have two issues:

1.  As mentioned above, inability to use relative paths makes it impossible to use the `.gitleaksignore` in a multi-agent environment.  For example, the fingerprint for one secret is `C:\azdo_winagent_1\_work\7\s\Certificates\Root-A.cer:CSCAN0020:2`, but it could be `C:\azdo_winagent_[1-5]\...` depending on how busy the agent machine is.

1. Running the Gitleaks from the decorator creates a `.json` file at the root of my project at compile-time, which is named using a random GUID.  For example, all my runs detect this secret:   `C:\azdo_winagent_1\_work\7\s\2376ab4a-328b-4141-b0f9-8ae379b199f4_39_2830.json:generic-api-key:1`

Contents of the auto-generated file are:
```
{"projectVitals":{"productName":"AD_Careers","name":"Security-Scans-Default","token":"***4a643b283d7a5996e35f2b0f73***774ff9b7bfc126******","creationDate":"2023-01-19 12:47:49","lastUpdatedDate":"2023-03-24 18:03:12"},"libraries":[]}
```
At face value I think support for wildcards in the `.gitleaksignore` could solve both of these problem.  

Any assistance would be appreciated.

**Comment by Devvox93** [comment]
I agree. Either supporting reletive paths or wildcards would solve al ot of issues with .gitleaksignore.
We're using hosted agents that are vm scale sets and the agents run jobs from many different projects/repo's.
The full path used in the fingerprint is, for example, `/agent/_work/4/s/`, where the 4 can be any number.

Is any work planned or ongoing on this? I wasn't able to find it in the issues list.

**Comment by xtermi2** [comment]
please support wildcards in `.gitleaksignore`. This prevents us from using it at the moment!
